rpcs3/util/asm.hpp

```diff
@@ -385,6 +385,13 @@ namespace utils
 		return static_cast<T>(value / align + (value > 0 ? T{(value % align) > (align / 2)} : 0 - T{(value % align) < (align / 2)}));
 	}
 
+	// Returns true if input is an unsigned integer with a power of 2
+	template <typename T> requires(std::is_integral_v<T> && std::is_unsigned_v<T>)
+	constexpr T is_power_of_2(T value)
+	{
+		return !(value & (value - 1));
```

Comment:
This also returns true if value == 0. Is that ok?

Reply:
Yes. The caller should handle 0 case. In practice, 0 is not a value you can do much with if you're filtering a quantity. I suppose I can add a comment.

Reply:
std::has_single_bit.